### PR TITLE
Add IP2Location as submodule

### PR DESCRIPTION
**Pull request by sgeto**
Maybe this seems like it's not really making any difference from the old way (batch file, `git clone` command, changing %include%, etc)
But on the long run, you will see that this is much more beneficial and can save a lot of work. (packaging, maintaining, integrating wsock_trace into other repositories).
The last one (integrating wsock_trace into other repositories) is especially important to me. This lib is a real gem and I use it all the time. Thanks for it btw.

P.S: If you like this PR as a patch file, [go here](https://patch-diff.githubusercontent.com/raw/gvanem/wsock-trace/pull/15.patch).

### CODE REVIEW

**Comment by sgeto** [comment]
You know what, **don't merge yet**. I have another idea...

File: src/Makefile.vc6 [hunk 1]
```diff
@@ -23,7 +23,7 @@ USE_DEF_FILE     = 1
 # '%INCLUDE%' variable instead of setting it here.
 #
 USE_IP2LOCATION  = 1
-IP2LOCATION_ROOT = C:\Users\Username\projects\IP2Location-C-Library
+IP2LOCATION_ROOT = $(MAKEDIR)\..\IP2Location-C-Library
```
Comment (gvanem) [inline src/Makefile.vc6]: `$(MAKEDIR)`? Some kind of Nmake built-in? Never heard of, but better than hard-coded paths.

**Comment by gvanem** [comment]
Okay, I'll hold the merge. This is a real cool patch. Remains to be tested.

File: src/Makefile.vc6 (hunk 1 above)
Comment (sgeto) [reply src/Makefile.vc6]: Yes. I found it deeply buried on [MSDN](https://msdn.microsoft.com/en-us/library/7cafx990.aspx). It resolves to the current directory.

**Comment by sgeto** [comment]
Don't worry I'll clean up the commits after I get what I am aiming at 😄

File: appveyor.yml [hunk 2]
```diff
@@ -1,27 +1,106 @@
...
+  matrix:
+    - BUILDER: msvc
+      CPU:     x86
+    - BUILDER: msvc
+      CPU:     x64
+    - BUILDER: mingw
+      CPU:     x86
+      MSYSTEM: MINGW32
+    - BUILDER: mingw
+      CPU:     x64
```
Comment (gvanem) [inline appveyor.yml]: Good to see 64-bit MinGW working with AppVeyor. But CygWin fails miserably 🤕. Oh well. I'll merge this later when I have the time. Cheers.

Comment (sgeto) [reply appveyor.yml]: hehe miserably indeed. 👍

**Comment by gvanem** [comment]
Merged https://github.com/gvanem/wsock-trace/commit/bc010c05d9adfcbcdd90af24a9a325047cbb3bc0
Thanks for your effort!

**Comment by sgeto** [comment]
> I disabled CygWin in AppVeyor for now

I understand. It has a long way to go...

**Pull request merged**